detox/src/utils/customConsoleLogger.js

```diff
@@ -0,0 +1,56 @@
+const util = require('util');
+const path = require('path');
+const callsites = require('./callsites');
+
+function callsitesToStack() {
+  return callsites().slice(2).reduce((result, callsite) => {
+    const filename = callsite.getFileName() ? path.relative(process.cwd(), callsite.getFileName()) : '';
+    const filePointer = filename ? `${filename}:${callsite.getLineNumber() || '?'}:${callsite.getColumnNumber() || '?'}` : '<unknown>';
+    const functionName = callsite.getFunctionName() || '<unknown>';
+    return result.concat(`\r    at ${functionName} (${filePointer})\n`)
+  }, '');
+}
+
+function getOrigin() {
+  const callsite = callsites()[2];
+  const filename = path.relative(process.cwd(), callsite.getFileName());
+  return `at ${filename}:${callsite.getLineNumber() || '?'}:${callsite.getColumnNumber() || '?'}`;
+}
+
+function override(consoleLevel, bunyanFn) {
+  console[consoleLevel] = (...args) => {
+    bunyanFn({ event: 'USER_LOG' }, getOrigin(), '\n', util.format(...args));
+  };
+}
+
+function overrideTrace(consoleLevel, bunyanFn) {
+  console[consoleLevel] = (...args) => {
+    bunyanFn({ event: 'USER_LOG' }, getOrigin(), '\n  Trace:', util.format(...args), '\n', callsitesToStack());
```

Comment:
Are you replicating Jest logger behavior here?

Reply:
Almost. It's similar but different.